src/etl/ETLService.cpp

```diff
@@ -278,11 +278,15 @@ ETLService::ETLService(
     , ledgerPublisher_(ioc, backend, backend->cache(), subscriptions, state_)
     , amendmentBlockHandler_(ioc, state_)
 {
-    startSequence_ = config.maybeValue<uint32_t>("start_sequence");
-    finishSequence_ = config.maybeValue<uint32_t>("finish_sequence");
-    state_.isReadOnly = config.valueOr("read_only", static_cast<bool>(state_.isReadOnly));
-    extractorThreads_ = config.valueOr<uint32_t>("extractor_threads", extractorThreads_);
-    txnThreshold_ = config.valueOr<size_t>("txn_threshold", txnThreshold_);
+    startSequence_ = config.getValue("start_sequence").hasValue()
+        ? std::make_optional(config.getValue("start_sequence").asIntType<uint32_t>())
+        : std::nullopt;
```

Comment:
`config.maybeValue<uint32_t>("...")` would be even better imo :)

Reply:
It seems to me that you could have kept the interface almost as it was yet implement all the fancy stuff inside of the framework. It would take much less changes to integrate the new config this way and the API would be more concise. Not saying we should change everything at this point but it's sad that we missed the opportunity there

Reply:
yea, looking back I think it would have better to kept the interface and changed the things that needed to be changed instead of implementing it from scratch 😕. But it was good practice on the bright side 👍